### PR TITLE
Introduce COMPlus_GDBJitElfDump

### PR DESCRIPTION
**Pull request by parjong**
This commit introduces COMPlus_GDBJitDump as a mean for debugging GDBJIT feature.

### CODE REVIEW

**Comment by parjong** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test please

**Comment by parjong** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test please

File: src/vm/gdbjit.cpp [hunk 1]
```diff
@@ -2473,12 +2487,35 @@ void NotifyGdb::OnMethodPrepared(MethodDesc* methodDescPtr)
...
+
+    if (!pszGDBJitElfDump)
+    {
+        LPWSTR pszGDBJitElfDumpW = NULL;
+        CLRConfig::GetConfigValue(CLRConfig::INTERNAL_GDBJitElfDump, &pszGDBJitElfDumpW);
+        pszGDBJitElfDump = EEConfig::NarrowWideChar(pszGDBJitElfDumpW);
+    }
+
+    LPCUTF8 methodName = methodDescPtr->GetName();
+    if (EEConfig::RegexOrExactMatch(pszGDBJitElfDump, methodName))
```
Comment (janvorli) [inline src/vm/gdbjit.cpp]: I would prefer adding ShouldDumpElfForMethod(LPCUTF8 methodName) and LPUTF8 pszGDBJitElfDump member to EEConfig to keep it in line with the other settings.

**Comment by janvorli** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test  please

**Pull request merged**